components/kyma-environment-broker/internal/process/deprovisioning/btp_operator_cleanup.go

```diff
@@ -0,0 +1,195 @@
+package deprovisioning
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/broker"
+	kebError "github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/error"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/provisioner"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/sirupsen/logrus"
+	k8serrors2 "k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	btpOperatorGroup           = "services.cloud.sap.com"
+	btpOperatorApiVer          = "v1"
+	btpOperatorServiceInstance = "ServiceInstance"
+	btpOperatorBinding         = "ServiceBinding"
+)
+
+type BTPOperatorCleanupStep struct {
+	operationManager  *process.DeprovisionOperationManager
+	provisionerClient provisioner.Client
+	k8sClientProvider func(kcfg string) (client.Client, error)
+}
+
+func NewBTPOperatorCleanupStep(os storage.Operations, provisionerClient provisioner.Client, k8sClientProvider func(kcfg string) (client.Client, error)) *BTPOperatorCleanupStep {
+	return &BTPOperatorCleanupStep{
+		operationManager:  process.NewDeprovisionOperationManager(os),
+		provisionerClient: provisionerClient,
+		k8sClientProvider: k8sClientProvider,
+	}
+}
+
+func (s *BTPOperatorCleanupStep) Name() string {
+	return "BTPOperator_Cleanup"
+}
+
+func (s *BTPOperatorCleanupStep) Run(operation internal.DeprovisioningOperation, log logrus.FieldLogger) (internal.DeprovisioningOperation, time.Duration, error) {
+	if !operation.Temporary {
+		log.Info("cleanup executed only for suspensions")
+		return operation, 0, nil
+	}
+	if operation.ProvisioningParameters.PlanID != broker.TrialPlanID {
+		log.Info("cleanup executed only for trial plan")
+		return operation, 0, nil
+	}
+	if operation.RuntimeID == "" {
+		log.Info("instance has been deprovisioned already")
+		return operation, 0, nil
+	}
+
+	kclient, err := s.getKubeClient(operation, log)
+	if err != nil {
+		return s.retryOnError(operation, err, log, "failed to get kube client")
```

Comment:
may be worth to run only `handleError` because cleaning up finalizers likely doesn't matter?

Reply:
after discussing the matter, we decided to keep retryOnError